api/src/main/java/org/apache/iceberg/ManifestFile.java

```diff
@@ -49,7 +49,7 @@ public interface ManifestFile {
           Types.LongType.get(),
           "Lowest sequence number in the manifest");
   Types.NestedField SNAPSHOT_ID =
-      optional(
```

Comment:
I think optional is correct here. 
V1 uses this then as optional in [V1Metadata](https://github.com/apache/iceberg/blob/9907a97351138e676d08e02ea53b746f4e331ec6/core/src/main/java/org/apache/iceberg/V1Metadata.java#L39).
V2 makes this then required in [V2Metadata](https://github.com/apache/iceberg/blob/9907a97351138e676d08e02ea53b746f4e331ec6/core/src/main/java/org/apache/iceberg/V2Metadata.java#L42)